cirq/devices/two_qubit_noise_model.py

```diff
@@ -0,0 +1,840 @@
+               self._p_iz, self._p_ix, self._p_iy
+
+    def __repr__(self) -> str:
+        return 'two_qubit_asymmetric_depolarize' \
+               '(p_xi={!r},p_xx={!r},p_xy={!r},p_xz={!r}, ' \
+               'p_yi={!r},p_yx={!r},p_yy={!r},p_yz={!r}, ' \
+               'p_zi={!r},p_zx={!r},p_zy={!r},p_zz={!r}, ' \
+               'p_ix={!r},p_iy={!r},p_iz={!r})'.format(
+            self._p_xi, self._p_xx, self._p_xy, self._p_xz,
+            self._p_yi, self._p_yx, self._p_yy, self._p_yz,
+            self._p_zi, self._p_zx, self._p_zy, self._p_zz,
+            self._p_iz, self._p_ix, self._p_iy)
+
+    def __str__(self) -> str:
+        return 'two_qubit_asymmetric_depolarize' \
+               '(p_xi={!r},p_xx={!r},p_xy={!r},p_xz={!r}, ' \
+               'p_yi={!r},p_yx={!r},p_yy={!r},p_yz={!r}, ' \
+               'p_zi={!r},p_zx={!r},p_zy={!r},p_zz={!r}, ' \
+               'p_ix={!r},p_iy={!r},p_iz={!r})'.format(
+            self._p_xi, self._p_xx, self._p_xy, self._p_xz,
+            self._p_yi, self._p_yx, self._p_yy, self._p_yz,
+            self._p_zi, self._p_zx, self._p_zy, self._p_zz,
+            self._p_iz, self._p_ix, self._p_iy)
+
+    def _circuit_diagram_info_(self,
+                               args: protocols.CircuitDiagramInfoArgs) -> str:
+        if args.precision is not None:
+            f = '{:.' + str(args.precision) + 'g}'
+            return 'A({},{},{},{},{},{},{},{},{},{},{},{},{},{},{},{})' \
+                .format(f, f, f, f, f, f, f, f, f, f, f, f, f, f, f, f, ).format(
+                        self._p_xi, self._p_xx, self._p_xy, self._p_xz,
+                        self._p_yi, self._p_yx, self._p_yy, self._p_yz,
+                        self._p_zi, self._p_zx, self._p_zy, self._p_zz,
+                        self._p_iz, self._p_ix, self._p_iy)
+        return 'A({!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r})'.format(
+            self._p_xi, self._p_xx, self._p_xy, self._p_xz, self._p_yi,
+            self._p_yx, self._p_yy, self._p_yz, self._p_zi, self._p_zx,
+            self._p_zy, self._p_zz, self._p_iz, self._p_ix, self._p_iy)
+
+
+def two_qubit_asymmetric_depolarize(p_xi: float, p_yi: float, p_zi: float,
+                                    p_xx: float, p_yx: float, p_zx: float,
+                                    p_xy: float, p_yy: float, p_zy: float,
+                                    p_xz: float, p_yz: float, p_zz: float,
+                                    p_ix: float, p_iy: float, p_iz: float
+                                   ) -> TwoQubitAsymmetricDepolarizingChannel:
+    return TwoQubitAsymmetricDepolarizingChannel(p_xi, p_yi, p_zi, p_xx, p_yx,
+                                                 p_zx, p_xy, p_yy, p_zy, p_xz,
+                                                 p_yz, p_zz, p_ix, p_iy, p_iz)
+
+
+# noinspection PyProtectedMember
+@cirq.value.value_equality
+class TwoQubitDepolarizingChannel(gate_features.TwoQubitGate):

```

Comment:
Let's move this to `cirq/ops`

Reply:
Does this include the asymmetric depolarising channel and all of the associated gates here?

Reply:
Yes. Channels should also be in `ops`.